Jack/Property.fs

```diff
@@ -337,27 +355,43 @@ module Property =
 
 [<AutoOpen>]
 module PropertyBuilder =
+    let rec private loop (p : unit -> bool) (m : Property<unit>) : Property<unit> =
+        if p () then
+            Property.bind m (fun _ -> loop p m)
+        else
+            Property.success ()
 
     type Builder internal () =
         member __.For(m : Property<'a>, k : 'a -> Property<'b>) : Property<'b> =
             Property.bind m k
 
         member __.For(xs : seq<'a>, k : 'a -> Property<unit>) : Property<unit> =
-            let join m n =
-                Property.bind m (fun _ -> n)
-            let init =
-                Property.success ()
-            xs |> Seq.map k |> Seq.fold join init
+            let xse = xs.GetEnumerator ()
+            Property.using xse <| fun xse ->
+                let mv = xse.MoveNext
+                let kc = Property.delay (fun () -> k xse.Current)
+                loop mv kc
```

Comment:
Yeah I was thinking maybe `Random` should be `Seed -> Size -> Trampoline<'a>`, will do some experimentation when I get a chance